IBX-3465: Added Nginx vhost template

glye opened the pull request:
| Question      | Answer
| ------------- | ---
| JIRA Ticket   | https://issues.ibexa.co/browse/IBX-3465
| Versions      | All supported

Like https://github.com/ezsystems/developer-documentation/pull/1466 but for Nginx.
Sample vhost copied from https://github.com/ibexa/docker/blob/main/templates/nginx/vhost.template and Ibexified.

NB: It refers to files in ez_params.d, meaning in https://github.com/ibexa/docker/tree/main/templates/nginx/ez_params.d
Should we copy them over, too?

Alternatively, should we just link directly to the vhost file in the docker repo and not copy anything over?

DominikaK commented on docs/getting_started/install_ez_platform.md:
```diff
@@ -317,6 +317,10 @@ Prepare a [virtual host configuration](https://httpd.apache.org/docs/2.4/vhosts/
 
+!!! note "Nginx"
+
+    If you are using Nginx instead of Apache, please use  [this example vhost file](https://github.com/ezsystems/developer-documentation/tree/master/code_samples/install/vhost_template/nginx-vhost.template) and modify it to fit your project. Also modify the instructions below accordingly.
```
```suggestion
    If you are using Nginx instead of Apache, use [this example vhost file](https://github.com/ezsystems/developer-documentation/tree/master/code_samples/install/vhost_template/nginx-vhost.template) and modify it to fit your project. Also modify the instructions below accordingly.
```

DominikaK reviewed (commented):
The rationale for copying the files here was that the docker repo is unsupported, so I guess copying both files makes most sense.

glye commented:
In that case, should we copy over the 6 extra params files, as well?

DominikaK commented:
Copying over is imperfect, ideally we should have a separate repo with this (I guess there could be other "resources" like this for the whole system), and keeping it in doc risks that the files will not be updated.

glye commented:
What say ye, @mnocon and @micszo? It does sound nice to me to maintain default vhosts and include them in testing, so they will be updated.

micszo commented:
What about ibexa/post-install? Why not there?

glye commented:
So adding "apache" and "nginx" dirs with vhost examples under "resources" in ibexa/post-install? Sounds good to me. Ref. https://github.com/ibexa/post-install/tree/main/resources

What do the package maintainers there say about that? @Nattfarinn @webhdx @adamwojs

glye commented:
Post-release ping :)

> What do the package maintainers there say about that? @Nattfarinn @webhdx @adamwojs

webhdx commented:
It makes sense to move them to ibexa/post-install. We should also make a command for generating vhosts. I remember we had a bash script back in a day: https://github.com/ezsystems/ezplatform/blob/master/bin/vhost.sh

glye commented:
Thanks. I copied the templates in https://github.com/ibexa/post-install/pull/50
If that passes, I'll upgrade this PR to make use of them.

glye commented:
It's necromancy time! 🦹‍♂️💀
@DominikaK, since https://github.com/ibexa/post-install/pull/50 is merged, post-install now contains both apache and nginx vhosts in https://github.com/ibexa/post-install/tree/main/resources/templates and they are supposed to be maintained there.

So it makes sense to
1. remove my nginx-vhost.template from this PR
2. maybe also remove the existing apache vhost.template (or not, optional)
3. update my changes to install_ibexa_dxp.md such that both the apache and nginx sections refer to the files in post-install

Agree?

DominikaK commented:
@glye: Yes, Almost yes and Yes :)

I'm all for removing the template file from this repo, but it's also referred to in https://github.com/ezsystems/developer-documentation/blob/master/docs/infrastructure_and_maintenance/environments.md and https://github.com/ezsystems/developer-documentation/blob/master/docs/infrastructure_and_maintenance/security/security_checklist.md. Can you update links in those two as well?

glye commented:
Bunch of changes:

- Split install doc into Apache and nginx sections.
- Standardised on the "nginx" all lower case spelling which is used above in the doc, and seems to be official.
- Removed both Apache and nginx samples from doc, and refer both to those in post-install.
- Clarify instructions, you don't have to hardcode doc root and app env.
- Also pointed Environments and Security Checklist pages to post-install vhost.
- Replaced ref to line number with a text string, so it doesn't break on changes.

DominikaK commented:
I've added one commit moving the two sections, Apache and nginx, into tabs for better viewing.

glye commented:
@DominikaK Right, that sounds good. 👍
